Issue#7 solution "Implement GET /methods/ API "

RomanZorkin opened the pull request:
Changes have been made to the project to implement the GET /methods/ API:
1.	In the module api.py added the import string of the local module "methods_list" and created a new function " def methods ():". It returns a list of available methods supported by geopatterns in the specified API format;
2.	The methods_list.py module was created. The module generates a list of methods of the geopatterns library, the Enum library is used for processing.

RomanZorkin committed:
Changes have been made to the project to implement the GET /methods/ API:
1.	In the module api.py added the import string of the local module "methods_list" and created a new function " def methods ():". It returns a list of available methods supported by geopatterns in the specified API format;
2.	The methods_list.py module was created. The module generates a list of methods of the geopatterns library, the Enum library is used for processing.

RomanZorkin committed:
Changes have been made to the project to implement the GET /methods/ API:
1.	In the module api.py added the import string of the local module "methods_list" and created a new function " def methods ():". It returns a list of available methods supported by geopatterns in the specified API format;
2.	The methods_list.py module was created. The module generates a list of methods of the geopatterns library, the Enum library is used for processing.

RomanZorkin committed:
Update methods_list.py

Correct the last string

anatoly-scherbakov commented:
@RomanZorkin please merge current `master` branch into your copy of the project. There is a bug in `flake8` which I fixed at #11; that bug prevents the tests from running. Should be fine once you do the merge.

anatoly-scherbakov commented on geopatterns_demo/methods_list.py:
```diff
@@ -0,0 +1,13 @@
+        methods_list.append(
+            {
+                'name': name,
+                'label': label
```
While you are at your environment, please try running:

```bash
make lint
```

This will run the linters, they will complain about a few things. Need to satisfy them. In this occasion, they will likely complain on missing comma, which should be like this:

```suggestion
                'label': label,
```

anatoly-scherbakov commented on geopatterns_demo/methods_list.py:
```diff
@@ -0,0 +1,13 @@
+from geopatterns_demo.models import Method
+
+def list_creator():
```
```suggestion
def describe_available_methods():
```

anatoly-scherbakov commented on geopatterns_demo/methods_list.py:
```diff
@@ -0,0 +1,13 @@
+    """Generate list of the methods geopatterns supports."""
+    methods_list = []
+    for name, label in Method.__members__.items():
+        methods_list.append(
```
Can we do this without `.append()`? For example, via a list comprehension.

RomanZorkin committed:
Merge branch 'anatoly-scherbakov:master' into master

RomanZorkin committed:
Update api.py

RomanZorkin reviewed (commented):
Adjustments were made at the suggestion of the project owner

anatoly-scherbakov commented:
@RomanZorkin acknowledged. It seems GitHub does not automatically run tests on new PRs somehow. I have run them and here's what they print:

```
geopatterns_demo/api.py:11: error: Module 'geopatterns_demo.methods_list' has no attribute 'describe_available_methods'
Found 1 error in 1 file (checked 7 source files)
```

That is because the method `list_creator()` itself was not renamed.

RomanZorkin committed:
1. The name of the method has been adjusted to "describe_available_methods"
2. The list is formed using "list comprehension" method
3. There is an error (WPS609) in the process of applying "flake8". The error warns about the use of the hidden attribute __members__ of the Enum library, and suggests using the super () method. The library documentation assumes the use of this attribute in projects "The __members__ attribute can be used for detailed programmatic access to the enumeration members. For example, finding all the aliases"

anatoly-scherbakov commented:
@RomanZorkin okay, now linters execute and they are unhappy about a few things: https://github.com/anatoly-scherbakov/geopatterns-demo/pull/10/checks?check_run_id=3231681853

Please look into how to fix that. For example, I think you can use just `list(Method)` instead of `Method.__members__`.

RomanZorkin committed:
1. Fixed errors in accordance with the requirements of linters

RomanZorkin committed:
1. Fixed errors in accordance with the requirements of linters

RomanZorkin reviewed (commented):
Fixed errors in accordance with the requirements of linters
